File: calculator.rb
# Ruby - 除法计算
def divide(a, b)
  b != 0 ? a / b : Float::INFINITY
end

begin
  lines = File.readlines('input.txt')
  output = "Ruby除法:\n"
  
  lines.each_with_index do |line, index|
    numbers = line.strip.split.map(&:to_f)
    if numbers.length >= 2
      a, b = numbers[0], numbers[1]
      result = divide(a, b)
      output += "行#{index + 1}: #{a} ÷ #{b} = #{result.round(4)}\n"
    end
  end
  
  File.open('output.txt', 'a') do |f|
    f.puts output + "\n"
  end
  
  puts "Ruby除法计算完成!"
rescue => e
  puts "Ruby计算错误: #{e.message}"
end

File: calculator.kt
// Kotlin - 取模运算
import java.io.File

fun modulo(a: Int, b: Int): Int {
    return if (b != 0) a % b else 0
}

fun main() {
    try {
        val lines = File("input.txt").readLines()
        val output = StringBuilder("Kotlin取模运算:\n")
        
        lines.forEachIndexed { index, line ->
            val numbers = line.trim().split("\\s+".toRegex())
            if (numbers.size >= 2) {
                val a = numbers[0].toIntOrNull()
                val b = numbers[1].toIntOrNull()
                if (a != null && b != null && b != 0) {
                    val result = modulo(a, b)
                    output.append("行${index + 1}: $a % $b = $result\n")
                }
            }
        }
        
        File("output.txt").appendText(output.toString() + "\n")
        println("Kotlin取模运算完成!")
    } catch (e: Exception) {
        println("Kotlin计算错误: ${e.message}")
    }
}